core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/V4Context.java

```diff
@@ -15,15 +15,17 @@
 
 package software.amazon.awssdk.http.auth.aws.internal.signer;
 
-import software.amazon.awssdk.annotations.Immutable;
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.http.SdkHttpRequest;
 
 /**
  * A container for data produced during and as a result of the SigV4 request signing process.
  */
 @SdkInternalApi
-@Immutable
+// TODO(sra-identity-auth): This is currently not @Immutable because signedRequest is a Builder. Is Builder needed? If it could
+//  hold reference to SdkHttpRequest instead, this class would be @Immutable.
+// TODO(sra-identity-auth): Consider if we can rename this to convey something more. maybe,
+//  V4RequestSigningResult/V4RequestSigningResultData? Note there is V4aContext similarly.
```

Comment:
Yes, V4Context is a terrible name. I like `V4RequestSigningResult`  or `V4RequestSigningOutput`

Reply:
will take up the TODOs separately.

Reply:
yup, I noticed later it can be changed, but didn't take that on immediately. It's SdkInternal, so non-blocking, so was planing to address both TODOs later.